### PR TITLE
[FIX] external code CMakelists.txt contains outdated variables

### PR DESCRIPTION
**Pull request by cbielow**
The CmakeVar @CF_OPENMS_ADD_DEFINITIONS@ seems to have been removed a while ago.

This PR updates the CMakeLists.txt for the external code documentation, plus some minor typos etc.

### CODE REVIEW

File: share/OpenMS/examples/external_code/CMakeLists.txt [hunk 1]
```diff
@@ -25,34 +25,35 @@ if (OpenMS_FOUND)
...
   ## Warning: this could be harmful to your project. Check this if problems occur.
   ## Also, use this to add your own compiler flags, e.g. for OpenMP support.
   ## e.g. for Visual Studio use /openmp
-  ## set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OPENMS_ADDCXX_FLAGS} /openmp")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OPENMS_ADDCXX_FLAGS}")
-  add_definitions(${OPENMS_DEFINITIONS})
-
+  # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OPENMS_ADDCXX_FLAGS} /openmp")
+  
+  add_definitions(/DBOOST_ALL_NO_LIB) ## disable auto-linking of boost libs (Boost tends to guess wrong lib names)
```
Comment (jpfeuffer) [inline share/OpenMS/examples/external_code/CMakeLists.txt]: CMake will transform this platform-independent right?

Comment (cbielow) [inline share/OpenMS/examples/external_code/CMakeLists.txt]: yep. at least thats what their docs say. Might not work in very old CMake versions ...

**Comment by cbielow** [comment]
test pass. Merge?

**Comment by jpfeuffer** [comment]
Yes. +1

**Pull request merged**